app/src/main/java/com/lightricks/feedexercise/network/FeedApiService.kt

```diff
@@ -1,5 +1,36 @@
 package com.lightricks.feedexercise.network
 
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+
+
 /**
  * todo: add the FeedApiService interface and the Retrofit and Moshi code here
  */
+
+
+interface FeedApiService {
+    @GET("Android/demo/feed.json")
+    fun getFeed():  Single<MetadataItemsList>
+
+}
+
+object NetworkAccess {
+    private const val BASE_URL = "https://assets.swishvideoapp.com/"
+
+    private val retrofit: Retrofit = run {
+        val moshi: Moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
```

Comment:
We should be able to run the code without the `KotlinJsonAdapterFactory` that uses reflection to parse the objects from and to json. Try to remove it and see if your code is still working.

Reply:
when removing the code doesn't run